test/e2e/Tests/Tests/TerminateOrchestratorTests.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Net;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Azure.Durable.Tests.DotnetIsolatedE2E;
+
+[Collection(Constants.FunctionAppCollectionName)]
+public class TerminateOrchestratorTests
+{
+    private readonly FunctionAppFixture _fixture;
+    private readonly ITestOutputHelper _output;
+
+    public TerminateOrchestratorTests(FunctionAppFixture fixture, ITestOutputHelper testOutputHelper)
+    {
+        _fixture = fixture;
+        _fixture.TestLogs.UseTestLogger(testOutputHelper);
+        _output = testOutputHelper;
+    }
+
+    private static async Task AssertTerminateRequestFails(HttpResponseMessage terminateResponse)
+    {
+        Assert.Equal(HttpStatusCode.BadRequest, terminateResponse.StatusCode);
+
+        string? terminateResponseMessage = await terminateResponse.Content.ReadAsStringAsync();
+        Assert.NotNull(terminateResponseMessage);
+        Assert.Equal("Status(StatusCode=\"Unknown\", Detail=\"Exception was thrown by handler.\")", terminateResponseMessage);
+    }
+
+    private static async Task AssertTerminateRequestSucceeds(HttpResponseMessage terminateResponse)
+    {
+        Assert.Equal(HttpStatusCode.OK, terminateResponse.StatusCode);
+
+        string? terminateResponseMessage = await terminateResponse.Content.ReadAsStringAsync();
+        Assert.NotNull(terminateResponseMessage);
+        Assert.Empty(terminateResponseMessage);
+    }
+
+
+    [Fact]
+    public async Task TerminateRunningOrchestration_ShouldSucceed()
+    {
+        using HttpResponseMessage response = await HttpHelpers.InvokeHttpTrigger("LongOrchestrator_HttpStart", "");
+
+        Assert.Equal(HttpStatusCode.Accepted, response.StatusCode);
+        string instanceId = await DurableHelpers.ParseInstanceId(response);
+        string statusQueryGetUri = await DurableHelpers.ParseStatusQueryGetUri(response);
+
+        Thread.Sleep(1000);
+
+        var orchestrationDetails = await DurableHelpers.GetRunningOrchestrationDetails(statusQueryGetUri);
+        Assert.Equal("Running", orchestrationDetails.RuntimeStatus);
+
+        using HttpResponseMessage terminateResponse = await HttpHelpers.InvokeHttpTrigger("TerminateInstance", $"?instanceId={instanceId}");
+        await AssertTerminateRequestSucceeds(terminateResponse);
+
+        Thread.Sleep(1000);
+
+        orchestrationDetails = await DurableHelpers.GetRunningOrchestrationDetails(statusQueryGetUri);
+        Assert.Equal("Terminated", orchestrationDetails.RuntimeStatus);
+    }
+
+
+    // This test is likely exposing some unintended behavior. Currently, attempting to terminate scheduled orchestrations has no effect.
+    // If the behavior of Terminate is changed to accomodate terminating scheduled orchestrations, this test will need to be modified accordingly.
+    [Fact]
```

Comment:
Consider updating this test to do what we want it to do (terminate the pending orchestration), opening a bug to track the current buggy behavior, and then marking the test as being skippable until the bug is fixed. For example:
```suggestion
    [Fact(Skip = "Will enable when {link to GitHub issue} is fixed.")]
```

Reply:
Done